core/store/src/trie/prefetching_trie_storage.rs

```diff
@@ -267,10 +307,13 @@ impl PrefetchStagingArea {
 
     fn insert_fetched(&self, key: CryptoHash, value: Arc<[u8]>) {
         let mut guard = self.0.lock().expect(POISONED_LOCK_ERR);
-        guard.size_bytes -= PREFETCH_RESERVED_BYTES_PER_SLOT;
         guard.size_bytes += value.len();
-        let pending = guard.slots.insert(key, PrefetchSlot::Done(value));
-        debug_assert!(prefetch_state_matches(PrefetchSlot::PendingPrefetch, &pending.unwrap()));
+        if let Some(dropped) = guard.slots.insert(key, PrefetchSlot::Done(value)) {
+            guard.size_bytes -= Self::reserved_memory(dropped);
+        } else {
+            guard.metrics.prefetch_staged_items.inc();
```

Comment:
explain when this scenario happens.. as this means that we're fetched something that we didn't request ? (or at least it was not in the slots)